app/client/genericState/api.ts

```diff
@@ -0,0 +1,28 @@
+import { apiClient } from "../api/client/apiClient";
+import { ApiClientType } from "../api/client/types";
+
+export type GetRequest = Pick<ApiClientType, "url" | "query" | "path">;
+export type PutRequest = Pick<ApiClientType, "url" | "query" | "path" | "body">;
+export type PostRequest = Pick<ApiClientType, "url" | "query" | "body">;
+export const genericGetAPI = ({ url, query, path }: GetRequest) =>
+  apiClient({
+    method: "GET",
+    url: `/${url}/`,
+    query: query,
+    path: path
+  });
+export const genericPutAPI = ({ url, query, path, body }: PutRequest) =>
+  apiClient({
+    method: "PUT",
+    url: `/${url}/`,
```

Comment:
why not just `url: url`?

Reply:
Because I think the apiclient expects things urls of the form '/foobar/', at least what I could tell from other cases lol!